ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapper.java

```diff
@@ -0,0 +1,274 @@
+        CCDDefendant.CCDDefendantBuilder builder,
+        FullDefenceResponse fullDefenceResponse
+    ) {
+
+        builder.responseDefenceType(
+            CCDDefenceType.valueOf(fullDefenceResponse.getDefenceType().name())
+        );
+        builder.responseDefence(fullDefenceResponse.getDefence().orElse(EMPTY));
+        fullDefenceResponse.getPaymentDeclaration().ifPresent(mapPaymentDeclaration(builder));
+        fullDefenceResponse.getEvidence().ifPresent(mapDefendantEvidence(builder));
+        fullDefenceResponse.getTimeline().ifPresent(mapDefendantTimeline(builder));
+    }
+
+    private Consumer<DefendantTimeline> mapDefendantTimeline(CCDDefendant.CCDDefendantBuilder builder) {
+        return timeline -> {
+            builder.defendantTimeLineComment(timeline.getComment().orElse(EMPTY));
+            builder.defendantTimeLineEvents(asStream(timeline.getEvents())
+                .map(timelineEventMapper::to)
+                .filter(Objects::nonNull)
+                .map(event -> CCDCollectionElement.<CCDTimelineEvent>builder().value(event).build())
+                .collect(Collectors.toList())
+            );
+        };
+    }
+
+    private Consumer<DefendantEvidence> mapDefendantEvidence(CCDDefendant.CCDDefendantBuilder builder) {
+        return evidence -> {
+            builder.responseEvidenceComment(evidence.getComment().orElse(EMPTY));
+            builder.responseEvidenceRows(asStream(evidence.getRows())
+                .map(evidenceRowMapper::to)
+                .filter(Objects::nonNull)
+                .map(row -> CCDCollectionElement.<CCDEvidenceRow>builder().value(row).build())
+                .collect(Collectors.toList())
+            );
+        };
+    }
+
+    private Consumer<PaymentDeclaration> mapPaymentDeclaration(CCDDefendant.CCDDefendantBuilder builder) {
+        return paymentDeclaration -> {
+            builder.paymentDeclarationExplanation(paymentDeclaration.getExplanation());
+            builder.paymentDeclarationPaidDate(paymentDeclaration.getPaidDate());
+        };
+    }
+
+    public void from(Claim.ClaimBuilder claimBuilder, CCDDefendant defendant) {
+        requireNonNull(claimBuilder, "claimBuilder must not be null");
+        requireNonNull(defendant, "defendant must not be null");
+        if (defendant.getResponseType() == null) {
+            return;
+        }
+
+        switch (defendant.getResponseType()) {
+            case FULL_DEFENCE:
+                claimBuilder.response(extractFullDefence(defendant));
+                break;
+            case FULL_ADMISSION:
+                claimBuilder.response(extractFullAdmission(defendant));
+                break;
+            case PART_ADMISSION:
+                claimBuilder.response(extractPartAdmission(defendant));
+                break;
+            default:
+                throw new MappingException("Invalid responseType");
+        }
+    }
+
+    private FullDefenceResponse extractFullDefence(CCDDefendant defendant) {
```

Comment:
like what I said above, these method can be part of separate Mapper file.

Reply:
As discussed, not here.